src/script/lua_api/l_vmanip.cpp

```diff
@@ -110,14 +115,20 @@ int LuaVoxelManip::l_set_data(lua_State *L)
 
 int LuaVoxelManip::l_write_to_map(lua_State *L)
 {
-	GET_ENV_PTR;
-
 	LuaVoxelManip *o = checkObject<LuaVoxelManip>(L, 1);
 	bool update_light = !lua_isboolean(L, 2) || readParam<bool>(L, 2);
 
 	if (o->vm->isOrphan())
 		return 0;
 
+	// This wouldn't work anyway as we have no env ptr, but it's still unsafe.
+	if (getEmergeThread(L)) {
+		log_deprecated(L, "write_to_map called in mapgen environment");
```

Comment:
Why not error right away?

Reply:
dunno. I guess the idea was that this is just a minor modder mistake, not a fatal error.

Reply:
Now that we introduce this new mapgen env concept, we'd still have the liberty to be pedantic from the beginning and allow relaxing later if necessary.